providers/writing_a_new_provider.md

```diff
@@ -0,0 +1,213 @@
+create  spec/models/manageiq/providers/awesome_cloud
+  gsub  lib/tasks_private/spec.rake
+create  systemd/manageiq-providers-awesome_cloud_cloud_manager_event_catcher@.service
+create  systemd/manageiq-providers-awesome_cloud_cloud_manager_event_catcher.target
+create  systemd/manageiq-providers-awesome_cloud_cloud_manager_refresh@.service
+create  systemd/manageiq-providers-awesome_cloud_cloud_manager_refresh.target
+create  systemd/manageiq-providers-awesome_cloud_cloud_manager_metrics_collector@.service
+create  systemd/manageiq-providers-awesome_cloud_cloud_manager_metrics_collector.target
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/event_catcher/runner.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/event_catcher/stream.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/metrics_collector_worker/runner.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/refresh_worker/runner.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/event_catcher.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/metrics_capture.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/metrics_collector_worker.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/refresh_worker.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/refresher.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager/vm.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/collector.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/parser.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/persister.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/collector/cloud_manager.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/parser/cloud_manager.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/persister/definitions/cloud_collections.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/persister/cloud_manager.rb
+identical  app/models/manageiq/providers/awesome_cloud/inventory/persister.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager.rb
+insert  .yamllint
+append  spec/spec_helper.rb
+```
+
+That's a lot of stuff!  We'll cover all of it in detail later but for now lets take a look at our plugin:
+
+```bash
+$ ls plugins/
+manageiq-providers-awesome_cloud
+```
+
+The generator also automatically adds your plugin to your local Gemfile:
+
+```bash
+$ git diff
+diff --git a/Gemfile b/Gemfile
+index b7e6783821..685474c570 100644
+--- a/Gemfile
++++ b/Gemfile
+ # This default is used to automatically require all of our gems in processes that don't specify which bundler groups they want.
+ #
+ ### providers
++
++group :awesome_cloud, :manageiq_default do
++  manageiq_plugin "manageiq-providers-awesome_cloud" alphabetically...
```

Comment:
Something happened here during the copy-paste, and it's not valid Ruby here.

Reply:
Some weird line wrapping going on with my editor, not deleting the full line